mtools/test/test_mloginfo.py

```diff
@@ -400,6 +400,109 @@ def _test_rsinfo(self, logfile_path, **expected):
+        assert any(map(lambda line: 'no error messages found.' in line, lines))
+        assert any(map(lambda line: 'no chunk migrations found.' in line, lines))
+        assert any(map(lambda line: 'no chunk migrations found.' in line, lines))
+        assert any(map(lambda line: 'no chunk splits found.' in line, lines))
+
+    def test_sharding_overview_shard(self):
+        self._test_init('sharding_379_shard.log')
+        self.tool.run('--sharding %s' % self.logfile_path)
+        output = sys.stdout.getvalue()
+        lines = output.splitlines()
+
+        assert any(map(lambda line: '(shard)' in line, lines))
+        assert any(map(lambda line: 'shard01' in line, lines))
+        assert any(map(lambda line: 'localhost:27018,'
+                                    'localhost:27019,'
+                                    'localhost:27020' in line, lines))
+        assert any(map(lambda line: 'configRepl' in line, lines))
+        assert any(map(lambda line: 'localhost:27033' in line, lines))
+
+    def test_sharding_overview_csrs(self):
+        self._test_init('sharding_379_CSRS.log')
+        self.tool.run('--sharding %s' % self.logfile_path)
+        output = sys.stdout.getvalue()
+        lines = output.splitlines()
+
+        assert any(map(lambda line: '(CSRS)' in line, lines))
+        assert any(map(lambda line: 'shard01' in line, lines))
+        assert any(map(lambda line: 'localhost:27018,'
+                                    'localhost:27019,'
+                                    'localhost:27020' in line, lines))
+        assert any(map(lambda line: 'configRepl' in line, lines))
+        assert any(map(lambda line: '[ { _id: 0, host: "localhost:27033",' 
+                                    ' arbiterOnly: false, buildIndexes: true,'
+                                    ' hidden: false, priority: 1.0, tags: {},'
+                                    ' slaveDelay: 0, votes: 1 } ]' in line, lines))
+        
+    def test_sharding_overview_mongos(self):
+        self._test_init('sharding_379_mongos.log')
+        self.tool.run('--sharding %s' % self.logfile_path)
+        output = sys.stdout.getvalue()
+        lines = output.splitlines()
+
+        assert any(map(lambda line: '(mongos)' in line, lines))
+        assert any(map(lambda line: 'shard01' in line, lines))
+        assert any(map(lambda line: 'localhost:27018,'
+                                    'localhost:27019,'
+                                    'localhost:27020' in line, lines))
+        assert any(map(lambda line: 'configRepl' in line, lines))
+        assert any(map(lambda line: 'localhost:27033' in line, lines))
+
+    def test_sharding_error_messages_exists(self):
+        self._test_init('sharding_379_shard.log')
+        self.tool.run('--sharding --errors %s' % self.logfile_path)
+        output = sys.stdout.getvalue()
+        lines = output.splitlines()
+        assert len(list(filter(lambda line: re.match('^  \d .* ', line),
```

Comment:
This test needs a comment explaining what the regex is trying to match (eg example of matching output line)